internal/nwucp/handler/handler.go

```diff
@@ -57,22 +57,22 @@ func HandleDLNASTransport(n3ueSelf *context.N3UE, nasMsg *nas.Message) {
 
 		newGREName := fmt.Sprintf("%s-id-%d", n3ueSelf.N3ueInfo.GreIfaceName, n3ueSelf.N3ueInfo.XfrmiId)
 
-		var linkGRE netlink.Link
-		if linkGRE, err = gre.SetupGreTunnel(newGREName, n3ueSelf.TemporaryXfrmiName, n3ueSelf.UEInnerAddr.IP,
+		var linkGREs []netlink.Link
+		if linkGREs, err = gre.SetupGreTunnels(newGREName, n3ueSelf.TemporaryXfrmiName, n3ueSelf.UEInnerAddr.IP,
 			n3ueSelf.TemporaryUPIPAddr, pduAddress, n3ueSelf.TemporaryQosInfo); err != nil {
-			naslog.Errorf("Setup GRE tunnel %s Fail %+v", newGREName, err)
+			naslog.Errorf("Setup GRE tunnel %s Fail: %+v", newGREName, err)
 			return
 		}
 
-		// Add route
+		// Add routes
+		// TODO: here we not ensure that linkGREs[0] is defalt path (QFI = 1)
```

Comment:
should use map for linkGREs (use QFI as key), so that the default path link can be found be QFI=1.